### PR TITLE
feat(pci): add new volume type consumption addons

### PR DESCRIPTION
**Pull request by sachinrameshn**
| Question         | Answer
| ---------------- | ---
| Branch?          | feat/gridscale-ga-m1
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Feat MANAGER-13539
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/storages/blocks/add/add.controller.js [hunk 1]
```diff
@@ -177,8 +177,15 @@ export default class PciBlockStorageAddController {
...
 
   onRegionChange() {
+    const { region } = this.storage;
     this.displaySelectedRegion = true;
     this.selectedVolumeAddon = null;
+    const volumeAddonsIds = this.volumesAvailability.plans?.filter(
+      ({ regions }) => regions.find(({ name }) => name === region.name),
+    );
+    this.consumptionVolumeAddons = this.totalConsumptionVolumeAddons.filter(
+      ({ planCode }) => volumeAddonsIds.find(({ code }) => code === planCode),
```
Comment (darsene) [inline packages/manager/modules/pci/src/projects/project/storages/blocks/add/add.controller.js:187]: **niptick** : `.some` should be more appropriate as you check a boolean value

File: packages/manager/modules/pci/src/projects/project/storages/blocks/add/add.controller.js [hunk 2]
```diff
@@ -177,8 +177,15 @@ export default class PciBlockStorageAddController {
   }
 
   onRegionChange() {
+    const { region } = this.storage;
     this.displaySelectedRegion = true;
     this.selectedVolumeAddon = null;
+    const volumeAddonsIds = this.volumesAvailability.plans?.filter(
+      ({ regions }) => regions.find(({ name }) => name === region.name),
+    );
+    this.consumptionVolumeAddons = this.totalConsumptionVolumeAddons.filter(
```
Comment (darsene) [inline packages/manager/modules/pci/src/projects/project/storages/blocks/add/add.controller.js:186]: no need to check nullity as you've checked it for `this.volumesAvailability.plans?`

File: packages/manager/modules/pci/src/projects/project/storages/blocks/add/add.controller.js (hunk 1 above)
Comment (sachinrameshn) [reply packages/manager/modules/pci/src/projects/project/storages/blocks/add/add.controller.js]: Its done

**Review by darsene — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/storages/blocks/add/add.controller.js [hunk 3]
```diff
@@ -177,8 +177,15 @@ export default class PciBlockStorageAddController {
   }
 
   onRegionChange() {
+    const { region } = this.storage;
     this.displaySelectedRegion = true;
     this.selectedVolumeAddon = null;
+    const volumeAddonsIds = this.volumesAvailability.plans?.filter(
+      ({ regions }) => regions.find(({ name }) => name === region.name),
```
Comment (antonymarion) [inline packages/manager/modules/pci/src/projects/project/storages/blocks/add/add.controller.js:184]: some here too, (consistent with below)

**Review by antonymarion — APPROVED** [review]

**Pull request merged**